process/smartContract/process.go

```diff
@@ -1234,13 +1246,20 @@ func (sc *scProcessor) createVMInputWithAsyncCallBackAfterBuiltIn(
 		}
 
 		gasLimit = outAcc.OutputTransfers[0].GasLimit
-		function, args, err := sc.argsParser.ParseCallData(string(outAcc.OutputTransfers[0].Data))
-		log.LogIfError(err, "function", "createVMInputWithAsyncCallBackAfterBuiltIn.ParseCallData")
-		if len(function) > 0 {
-			arguments = append(arguments, []byte(function))
-		}
 
-		arguments = append(arguments, args...)
+		if sc.flagFixAsyncCallBackArgumentsParser.IsSet() {
+			args, err := sc.argsParser.ParseArguments(string(outAcc.OutputTransfers[0].Data))
+			log.LogIfError(err, "function", "createVMInputWithAsyncCallBackAfterBuiltIn.ParseArguments")
```

Comment:
why not returning error here? Otherwise, the code execution will simply append a nil args slice and jump to L1265

Reply:
would leave as it is - as it calls back the contract (asyncCallBack) with empty args and that's it.